game-core/src/main/java/games/strategy/util/PointFileReaderWriter.java

```diff
@@ -188,6 +188,69 @@ public static void writeOneToMany(final OutputStream sink, final Map<String, Lis
     return mapping;
   }
 
+  /**
+   * Writes the specified one-to-many mapping between names and (points, overflowToLeft) to the specified stream.
+   *
+   * @param sink The stream to which the name-to-points mappings will be written.
+   * @param mapping The name-to-points mapping to be written.
+   *
+   * @throws IOException If an I/O error occurs while writing to the stream.
+   */
+  public static void writeOneToManyPlacements(final OutputStream sink,
+      final Map<String, Tuple<List<Point>, Boolean>> mapping)
+      throws IOException {
+    checkNotNull(sink);
+    checkNotNull(mapping);
+
+    final StringBuilder out = new StringBuilder();
+    final Iterator<String> keyIter = mapping.keySet().iterator();
+    while (keyIter.hasNext()) {
+      final String name = keyIter.next();
+      out.append(name).append(" ");
+      final Collection<Point> points = mapping.get(name).getFirst();
+      final boolean overflowToLeft = mapping.get(name).getSecond();
+      final Iterator<Point> pointIter = points.iterator();
+      while (pointIter.hasNext()) {
+        final Point point = pointIter.next();
+        out.append(" (").append(point.x).append(",").append(point.y).append(")");
+        if (pointIter.hasNext()) {
+          out.append(" ");
+        }
+      }
+      out.append(" | overflowToLeft=" + overflowToLeft);
+      if (keyIter.hasNext()) {
+        out.append("\r\n");
+      }
+    }
+    write(out, sink);
+  }
+
+  /**
+   * Returns a map of the form String -> (points, overflowToLeft).
+   */
+  public static Map<String, Tuple<List<Point>, Boolean>> readOneToManyPlacements(final InputStream stream)
+      throws IOException {
+    checkNotNull(stream);
+
+    final Map<String, List<Point>> mapping = new HashMap<>();
+    final Map<String, Tuple<List<Point>, Boolean>> result = new HashMap<>();
+    try (Reader inputStreamReader = new InputStreamReader(new CloseShieldInputStream(stream), StandardCharsets.UTF_8);
+        LineNumberReader reader = new LineNumberReader(inputStreamReader)) {
+      @Nullable
+      String current = reader.readLine();
+      while (current != null) {
+        if (current.trim().length() != 0) {
+          final String[] s = current.split(" \\| ");
+          final Tuple<String, List<Point>> tuple = readMultiple(s[0], mapping);
+          final boolean overflowToLeft = s.length == 2 ? Boolean.parseBoolean(s[1].split("=")[1]) : false;
+          result.put(tuple.getFirst(), Tuple.of(tuple.getSecond(), overflowToLeft));
+        }
+        current = reader.readLine();
```

Comment:
A for loop of this scheme would be more appropriate:
```java
for (String current = reader.readLine(); current != null; current = reader.readLine()) {
  //code
}
```
Apart from that I'd rather use a Scanner instead of a LineNumberReader. It doesn't look like the line number is relevant here.

Reply:
EDIT: Of course a BufferedReader or something would work as well.
I just wanted to point out the line number feature is not needed.

Reply:
Same as streams. I think moving the whole class to use for loops with scanner should be done for all methods in this class for consistency.